### PR TITLE
Fix: Fix complex assignment

### PR DESCRIPTION
**Pull request by Liaojinghui**
This pr soves the range check bug in complex assignment.

### CODE REVIEW

**Comment by Liaojinghui** [comment]
This pr is kinda the same as https://github.com/neo-project/neo-devpack-dotnet/pull/828, but since the branch name has set to be related to incand dec, i open a new pr here.

**Comment by shargon** [comment]
> This pr is kinda the same as #828, but since the branch name has set to be related to incand dec, i open a new pr here.

What's your plan? merge both or choose one?

**Comment by Liaojinghui** [comment]
> What's your plan? merge both or choose one?

They fix the issues caused by the same reason, but they are fixing different issues. This one focus on complex assignment syntax, another fix the unary syntax. Both need to be merged.

**Comment by Liaojinghui** [comment]
@shargon this one and https://github.com/neo-project/neo-devpack-dotnet/pull/828 are critial bugs, please review ASAP

File: tests/Neo.Compiler.CSharp.UnitTests/UnitTest_Contract_ComplexAssign.cs [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+            _engine.Reset();
+            var result = _engine.ExecuteTestCaseStandard("unitTest_Right_Shift_Assign_UnChecked");
+            Assert.AreEqual(VMState.HALT, _engine.State);
+            var values = (Array)result.Pop();
+            Assert.AreEqual(unchecked(uint.MinValue >> 1), values[0].GetInteger());
+            Assert.AreEqual(unchecked(int.MinValue >> 1), values[1].GetInteger());
+        }
+
+
+    }
```
Comment (shargon) [inline tests/Neo.Compiler.CSharp.UnitTests/UnitTest_Contract_ComplexAssign.cs:122]:
```suggestion
    }
```

File: tests/Neo.Compiler.CSharp.UnitTests/UnitTest_Contract_ComplexAssign.cs [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        public void Test_LeftShiftAssign_UnChecked()
+        {
+            _engine.Reset();
+            var result = _engine.ExecuteTestCaseStandard("unitTest_Left_Shift_Assign_UnChecked");
+            Assert.AreEqual(VMState.HALT, _engine.State);
+            var values = (Array)result.Pop();
+            Assert.AreEqual(unchecked(uint.MaxValue << 1), values[0].GetInteger());
+            Assert.AreEqual(unchecked(int.MaxValue << 1), values[1].GetInteger());
+
+        }
```
Comment (shargon) [inline tests/Neo.Compiler.CSharp.UnitTests/UnitTest_Contract_ComplexAssign.cs:97]:
```suggestion
        }
```

File: tests/Neo.Compiler.CSharp.UnitTests/TestClasses/Contract_ComplexAssign.cs [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+            int value2 = int.MinValue;
+            unchecked
+            {
+                value1>>=1;
+                value2>>=1;
+            }
+            return (value1, value2);
+        }
+
+    }
```
Comment (shargon) [inline tests/Neo.Compiler.CSharp.UnitTests/TestClasses/Contract_ComplexAssign.cs:128]:
```suggestion
    }
```

**Review by shargon — APPROVED** [review]

**Pull request merged**